esp/bindings/http/platform/httpbinding.cpp

```diff
@@ -1823,25 +1835,95 @@ int EspHttpBinding::onGetVersion(IEspContext &context, CHttpRequest* request, CH
     StringBuffer srvQName;
     qualifyServiceName(context, service, NULL, srvQName, NULL);
     verxml.appendf("<VersionInfo><Service>%s</Service><Version>%.3f</Version></VersionInfo>", srvQName.str(), m_wsdlVer);
-    
+
     response->setContent(verxml.str());
     response->setContentType(HTTP_TYPE_APPLICATION_XML_UTF8);
     response->setStatus(HTTP_STATUS_OK);
     response->send();
     return 0;
 }
 
+void getServiceSchema(const char *serviceQName, const char *methodQName, double version, const char *nstr,
+                      const StringBuffer &xmlFilename,  Owned<IProperties> &params, EsdlXslTypeId EsdlXslType, StringBuffer &schema)
+{
+    Owned<IEsdlDefinition> esdlDef;
+    Owned<IEsdlDefinitionHelper> defHelper;
+
+    esdlDef.set(createEsdlDefinition(nullptr, nullptr));
+    defHelper.set(createEsdlDefinitionHelper());
+
+    std::string xmlFilePath(getCFD());
+    xmlFilePath.append("esdl_files").append(PATHSEPSTR).append(xmlFilename.str());
+    esdlDef->addDefinitionsFromFile(xmlFilePath.c_str());
+    unsigned optFlags{DEPFLAG_COLLAPSE|DEPFLAG_ARRAYOF};
+    Owned<IEsdlDefObjectIterator> structs = esdlDef->getDependencies(serviceQName, methodQName, ";", version, nullptr, optFlags );
+
+    StringBuffer xlstFile(getCFD());
+    xlstFile.append("xslt").append(PATHSEPSTR).append("esp_service_xml2xsd.xslt");
+
+    defHelper->loadTransform(xlstFile, params, EsdlXslType);
+    defHelper->toWSDL(*structs, schema, EsdlXslType, version, nullptr, nstr, optFlags);
```

Comment:
Instead of passing nullptr for the `opts` parameter, we'll need to pull the URL parameters from the request and pass those instead. Then will be the issue of how to deal with ecm elements tagged `[optional]`. Either we can remove that from the ecm if it would have no negative effect, or we could figure out if there is some other value to insert (for this code path) in the `IProperties* opts` parameter that the transform would recognize as the name for that 'optional'. My hunch is maybe an empty string, but I'll need to check the generated xml format of esdl.

Reply:
Passed the context request parameters to the function and used accordingly.